foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/AddressResolverConfig.java

```diff
@@ -109,9 +109,12 @@ public static AddressResolverOptions getAddressResover(String tag, Configuration
 
   private static List<String> getStringListProperty(Configuration configSource,
       List<String> defaultValue, String... keys) {
-    if (configSource == null) {
+    if (configSource == null && DynamicPropertyFactory.getBackingConfigurationSource() != null) {
```

Comment:
There are lot of duplicated code, we could extract a method for it.

Reply:
ok